clang/test/Sema/implicit-constexpr.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %clang_cc1 -verify=normal,both -std=c++23 %s
+// RUN: %clang_cc1 -verify=implicit,both -fimplicit-constexpr -std=c++23 %s
```

Comment:
We should also test at least C++17 mode and see how this interacts with features that are not allowed in `constexpr` functions in that language mode (i.e. destructors, operator `new`, `goto`).

Reply:
new test added for earlier standards too